qemu/tests/boot_order_check.py

```diff
@@ -29,34 +29,39 @@ def _get_device(devices, dev_id):
                 device_found = dev
                 break
             elif dev['class_info']['desc'] == 'PCI bridge':
-                device_found = _get_device(dev['pci_bridge']['devices'],
+                pci_bridge_devices = dev['pci_bridge'].get('devices')
+                if not pci_bridge_devices:
+                    continue
+                device_found = _get_device(pci_bridge_devices,
                                            dev_id)
                 if device_found:
                     break
         return device_found
 
-    def _get_pci_addr_by_devid(vm, dev_id):
-        dev_addr = ''
-        dev_addr_fmt = '%02d:%02d.%d'
-        pci_info = vm.monitor.info('pci', debug=False)
-        if isinstance(pci_info, list):
-            device = _get_device(pci_info[0]['devices'], dev_id)
-            if device:
-                dev_addr = dev_addr_fmt % (device['bus'],
-                                           device['slot'],
-                                           device['function'])
-        else:
-            # As device id in the last line of info pci output
-            # We need reverse the pci information to get the pci addr which is in the
-            # front row.
-            pci_list = str(pci_info).split("\n")
-            pci_list.reverse()
-            pci_info = " ".join(pci_list)
-            dev_match = re.search(nic_addr_filter % dev_id, pci_info)
-            if dev_match:
-                bus_slot_func = [int(i) for i in dev_match.groups()]
-                dev_addr = dev_addr_fmt % tuple(bus_slot_func)
-        return dev_addr
+    def _get_pci_addr_by_devid(dev_id):
+        def wrap_func():
```

Comment:
I think no need \"wrap_func()\".  just use it like 
pci_addr = utils_misc.wait_for(lambda: _get_pci_addr_by_devid(nic.device_id),
                                                 timeout=devices_load_timeout)